src/JWT.Extensions.AspNetCore/JwtAuthenticationEvents.cs

```diff
@@ -41,47 +42,47 @@ static JwtAuthenticationEvents()
                 "Error decoding JWT: {0}, returning failure");
         }
 
-        public Func<ILogger, AuthenticateResult> OnMissingHeader { get; set; } =
+        public Func<ILogger, Task<AuthenticateResult>> OnMissingHeader { get; set; } =
             logger =>
             {
                 _logMissingHeader(logger, null);
-                return AuthenticateResult.NoResult();
+                return Task.FromResult(AuthenticateResult.NoResult());
             };
 
-        public Func<ILogger, string, string, AuthenticateResult> OnIncorrectScheme { get; set; } =
+        public Func<ILogger, string, string, Task<AuthenticateResult>> OnIncorrectScheme { get; set; } =
             (logger, actualScheme, expectedScheme) =>
             {
                 _logIncorrectScheme(logger, actualScheme, expectedScheme, null);
-                return AuthenticateResult.NoResult();
+                return Task.FromResult(AuthenticateResult.NoResult());
             };
 
-        public Func<ILogger, string, AuthenticateResult> OnEmptyHeader { get; set; } = (logger, header) =>
+        public Func<ILogger, string, Task<AuthenticateResult>> OnEmptyHeader { get; set; } = (logger, header) =>
         {
             _logEmptyHeader(logger, null);
-            return AuthenticateResult.NoResult();
+            return Task.FromResult(AuthenticateResult.NoResult());
```

Comment:
qq: What's the difference between `AuthenticateResult.None` and `AuthenticateResult.NoResult()`? Would the former a bit more efficient?

Reply:
[`AuthenticateResult.None`](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Authentication.Abstractions/src/AuthenticateResult.cs#L50) is an instance boolean property, rather than a factory method.

[`AuthenticateResult.NoResult()`](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Authentication.Abstractions/src/AuthenticateResult.cs#L89) is a factory method that returns a [static _noResult](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Authentication.Abstractions/src/AuthenticateResult.cs#L14) which should avoid an allocation.

So, NoResult appears to be the correct choice, and the ASP.Net team has already done us a solid in helping to keep things efficient. :)